File: __OTHER_FRAMEWORKS__/grafana/dashboards-as-code-workshop-main/part-one-python-starter/src/manifests.py
from grafana_foundation_sdk.models.dashboard import Dashboard
from grafana_foundation_sdk.models.resource import (
    Manifest as SDKManifest,
    Metadata,
)


class Manifest:
    @classmethod
    def dashboard(cls, folder_uid: str, dash: Dashboard) -> SDKManifest:
        if dash.uid is None:
            raise RuntimeError("dashboards must have a uid")

        return SDKManifest(
            api_version="dashboard.grafana.app/v1beta1",
            kind="Dashboard",
            metadata=Metadata(
                annotations={"grafana.app/folder": folder_uid},
                name=dash.uid,
            ),
            spec=dash,
        )

File: __OTHER_FRAMEWORKS__/grafana/dashboards-as-code-workshop-main/part-one-typescript-starter/src/manifests.ts
import { Dashboard } from "@grafana/grafana-foundation-sdk/dashboard";
import { Manifest } from "@grafana/grafana-foundation-sdk/resource";

export const dashboardManifest = (folderUid: string, dashboard: Dashboard): Manifest => {
    return {
        apiVersion: 'dashboard.grafana.app/v1beta1',
        kind: 'Dashboard',
        metadata: {
            annotations: {
                'grafana.app/folder': folderUid,
            },
            name: dashboard.uid!,
        },
        spec: dashboard,
    };
};
